.azure/pr.yml

```diff
@@ -0,0 +1,52 @@
+pr:
+  autoCancel: true
+  branches:
+    include: ["*"]
+
+stages:
+  # The preliminary stage should be small in both total runtime (including
+  # provisioning) and resources required.  About half of PR commits result in a
+  # CI failure, and over 90% of these are in linting, documention or a test
+  # failure that would affect _any_ OS or Python version.  The goal in the first
+  # stage is to catch the vast majority of failures with minimal cost.
+  - stage: "Lint_Docs_Prelim_Tests"
+    jobs:
+      - template: "templates/lint-linux.yml"
+        parameters:
+          pythonVersion: "3.7"
```

Comment:
I wonder if it'd be worth having a global variable template (e.g. `vars.yml`) imported by all pipelines containing things like `MIN_PYTHON_VERSION` (string), `SUPPORTED_PYTHON_VERSIONS` (list of strings), etc. so that we don't need to update versions in more than one place when we change what's supported.

https://docs.microsoft.com/en-us/azure/devops/pipelines/process/templates?view=azure-devops#variable-reuse

Reply:
Ohhh, and use them in the pipeline descriptions not the templates... Yeah that's a good idea - I originally dismissed it because I didn't want to hide the versions away within the templates themselves, but importing them into the pipelines and passing them in that way is a great idea.

Reply:
Done in 51884294f